bliss/surveys/dc2.py

```diff
@@ -30,16 +31,28 @@ def map_nested_dicts(cur_dict, func):
 
 
 class DC2Dataset(Dataset):
-    def __init__(self, split_files_list) -> None:
+    def __init__(self, split_files_list, in_memory: bool) -> None:
```

Comment:
I'm surprised this in-memory stuff is necessary. Is the DC2 data being staged to a local SSD? Seems like that should provide fast-enough access

Reply:
it's really strange that when i ran the exp last night, one epoch takes 30 min, but when i ran the exp with the same setting and refactored code, it takes me about 1h per epoch

Reply:
the data are stored at `/scrach` folder

Reply:
Is `/scratch` the local SSD?

Reply:
On the Great Lakes login nodes, `/scratch` is a massive network filesystem:
```
df -h | grep scratch
gl-scratch                                                          1.9P  1.5P  344T  82% /gpfs
```
This will be relatively slow to access.

Reply:
sorry, i make a stupid mistake. i'm now trying to copy data first to `/tmp` and then run the code

Reply:
There's supposed to be a 14 TB NVMe SSD on these SPGPU instances, which should be very fast. That would be a great place to stage the DC2 data.

Reply:
It appears to be mounted at `/tmp_data`.

Reply:
i find it. i'm now copying the data

Reply:
I recommend adding the copy command to your (non-interactive) bash login script because the data may not always be in `/tmp_data`; slurm can assign you to any SPGPU node and `/tmp_data` likely gets deleted occasionally. `rsync` may be even better than `cp` here: it copies only what is needed to make both versions of the data consistent.